### PR TITLE
feat(cc-logs-application-view): allow store and restore of the selected date range

### LINKED ISSUE #1074: cc-logs-application-view: dispatch event when date range selection changes
Context: We want to be able to store and restore the date range selection set by the user.

The first thing to do is to dispatch an event when the date range selection changes.

As of now, an event is dispatched when the date range changes. It's ok, but with that we only have the date range {since/until}. Now we want to have the date range selection which is an object like: `{type: 'live'}` or `{type:'predefined', def: 'lastHour'}` ...

### LINKED ISSUE #1075: cc-logs-application-view: allow initialization from a date range selection passed into the smart context
Context: We want to be able to store and restore the date range selection set by the user.

When a date range selection is passed into the smart context, we should initialize the view with that date range selection.

### PR DESCRIPTION
**Pull request by pdesoyres-cc**
Fixes #1074
Fixes #1075

- You should see a new event when changing the date range
- Use the sandbox with the dateRangeSelection selector

### CODE REVIEW

File: src/components/cc-logs-application-view/cc-logs-application-view.types.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { GhostInstance, Instance } from "../cc-logs-instances/cc-logs-instances.types";
-import { DateDisplay } from "../cc-logs/date-display.types";
-import { LogsControlPalette } from "../cc-logs-control/cc-logs-control.types";
-import { Timezone } from "../../lib/date/date.types";
+import { GhostInstance, Instance } from "../cc-logs-instances/cc-logs-instances.types.js";
+import { DateDisplay } from "../cc-logs/date-display.types.js";
+import { LogsControlPalette } from "../cc-logs-control/cc-logs-control.types.js";
+import { Timezone } from "../../lib/date/date.types.js";
```
Comment (hsablonniere) [inline src/components/cc-logs-application-view/cc-logs-application-view.types.d.ts:4]: praise: nice catch

**Review by hsablonniere — APPROVED** [review]
:+1: Let's go with this, I'll review the whole smart file later in the general logs review.

File: src/components/cc-logs-application-view/cc-logs-application-view.smart.js [hunk 2]
```diff
@@ -336,6 +340,19 @@ class LogsApplicationViewSmartController {
 
   }
 
+  /**
+   *
```
Comment (florian-sanders-cc) [inline src/components/cc-logs-application-view/cc-logs-application-view.smart.js:344]: **nitpick:** 
```suggestion

```

**Review by florian-sanders-cc — APPROVED** [review]
Apart from a very small nitpick, it looks good to me (from what I understand at least)!

**Review by HeleneAmouzou — APPROVED** [review]

**Review by roberttran-cc — APPROVED** [review]

**Pull request merged**